app/src/test/java/com/eveningoutpost/dexdrip/Models/BlueReaderTest.java

```diff
@@ -0,0 +1,556 @@
+        // :: Setup
+        byte[] buffer = "HYBERNATE SUCCESS".getBytes();
+
+        // :: Act
+        byte[] reply = blueReader.decodeblueReaderPacket(buffer, -1);
+
+        // :: Verify
+        assertThat(reply).isNull();
+        assertThat(getLogs()).contains("I/blueReader: blueReader notice that NFC is now really hibernated...");
+    }
+
+    @Test
+    public void decodeBlueReaderPacket_HybernateSuccess_HibernatedCounterNotZero() {
+        // :: Setup
+        byte[] buffer = "HYBERNATE SUCCESS".getBytes();
+        blueReader.decodeblueReaderPacket("not ready for".getBytes(), -1);
+
+        // :: Act
+
+        byte[] reply = blueReader.decodeblueReaderPacket(buffer, -1);
+
+        // :: Verify
+        assertThat(reply).isNotNull();
+        assertThat(new String(reply)).isEqualTo("l");
+        String log = getLogs();
+        assertThat(log).contains("I/blueReader: blueReader notice that NFC is now really hibernated...");
+        assertThat(log).contains("W/blueReader: Found hibernation after wrong read. Resend read-command...");
+    }
+
+    @Test
+    public void decodeBlueReaderPacket_Unknown() {
+        // :: Setup
+        byte[] buffer = "-r 0:ASDF".getBytes();
+
+        // :: Act
+        byte[] reply = blueReader.decodeblueReaderPacket(buffer, -1);
+
+        // :: Verify
+        assertThat(reply).isNull();
+        assertThat(getLogs()).contains("D/blueReader: blueReader sends an unknown reaktion: '-r 0:ASDF'");
+    }
+
+    @Test
+    public void decodeBlueReaderPacket_Battery_NoLastBgReading() {
+        // :: Setup
+        byte[] buffer = "battery: 222".getBytes();
+
+        // :: Act
+        byte[] reply = blueReader.decodeblueReaderPacket(buffer, -1);
+
+        // :: Verify
+        assertThat(reply).isNotNull();
+        assertThat(new String(reply)).isEqualTo("l");
+    }
+
+    @Test
+    public void decodeBlueReaderPacket_Battery_OldBgReading() {
+        // :: Setup
+        byte[] buffer = "battery: 4000 980".getBytes();
+
+        // Add mock bg readings
+        Sensor mockSensor = createMockSensor();
+        addMockBgReading(125, 10, mockSensor);
+
+        // :: Act
+        byte[] reply = blueReader.decodeblueReaderPacket(buffer, -1);
+
+        // :: Verify
+        assertThat(reply).isNotNull();
+        assertThat(new String(reply)).isEqualTo("l");
+    }
+
+    @Test
+    public void decodeBlueReaderPacket_Battery_NewBgReading() {
+        // :: Setup
+        byte[] buffer = "battery: 4000 980".getBytes();
+
+        // Add mock bg readings
+        Sensor mockSensor = createMockSensor();
+        addMockBgReading(125, 2, mockSensor);
+
+        // :: Act
+        byte[] reply = blueReader.decodeblueReaderPacket(buffer, -1);
+
+        // :: Verify
+        assertThat(reply).isNull();
+    }
+
+    @Test
+    public void decodeBlueReaderPacket_IDR_noBgReading() {
+        // :: Setup
+        String inputString = "IDR0blue131-a1";
```

Comment:
not sure what should be tested here, but normaly the inputstring should be with a '|' (IDR0|blue131-a1).